### PR TITLE
dev-libs/gumbo: add 0.12.1

### PR DESCRIPTION
**Pull request by lanodan**
Signed-off-by: Haelwenn (lanodan) Monnier <contact@hacktivis.me>

### CODE REVIEW

**Comment by kuraga** [comment]
This version is needed for `net-mail/claws-mail-{3.20,4.2}`.

**Comment by thesamesam** [comment]
If you're going to file bugs for them (which is fine), please then mention them on pre-existing PRs so that they can be updated to tag said bugs, so we don't end up with them remaining open.

**Comment by kuraga** [comment]
Sure (didn't know whether it is automated: I've added this PR to the bug).

Closes: https://bugs.gentoo.org/917743

File: dev-libs/gumbo/gumbo-0.12.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+inherit autotools
+
+DESCRIPTION="The HTML5 parsing algorithm implemented as a pure C99 library"
+HOMEPAGE="https://codeberg.org/grisha/gumbo-parser"
+SRC_URI="https://codeberg.org/grisha/gumbo-parser/archive/${PV}.tar.gz -> ${P}.tar.gz"
+S="${WORKDIR}/gumbo-parser"
+
+LICENSE="Apache-2.0"
+SLOT="0"
```
Comment (thesamesam) [inline dev-libs/gumbo/gumbo-0.12.1.ebuild:14]: The SONAME changes between 0.11.x and 0.12.x, and this is noted in `CHANGES.md`:
```
+## Gumbo 0.12.0 (2023-07-18)
+
+* Add support for `<dialog>` tag
+* Fix `TAGSET_INCLUDES` macro to work properly with multiple bit flags ([17ab1d2@lua-gumbo](https://github.com/craigbarnes/lua-gumbo/commit/17ab1d2b4c4742da78c0c8b3329f61a95744c895))
+* Avoid potential crash for zero length tags ([01359a9@nokogiri](https://github.com/sparklemotion/nokogiri/commit/01359a951e6c43bd975dfdb1fa80c01d32cc77d2))
+* Bump shared library version
+
```

Hence we need to add a subslot and make sure consumers are using it. I'll do that this time. See https://devmanual.gentoo.org/general-concepts/slotting/index.html#abi-breakage.

**Pull request closed without merging**